discretisedfield/tests/test_field.py

```diff
@@ -1725,10 +1729,11 @@ def test_write_read_ovf(self):
 
                     assert f.allclose(f_read)
                     assert f_read.units == "A/m"
+                    assert f.mesh.subregions == f_read.mesh.subregions
```

Comment:
It would be also good to test in all the `test_read_write_*` functions that if one defines `save_subregions=False`, and even if the `subregions` are defined, the `json` file is not being created.